proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,197 @@
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of members to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing members from.
+  string page_token = 4;
+}
+
+// Response to list members.
+message ListMembersResponse {
+  // ID of the group owner.
+  string owner_account_id = 1;
+
+  // List of members of the group.
+  map<string, Role> members = 2;
+
+  // Token to continue listing members from.
+  string next_page_token = 3;
+}
+
+// Request to list content.
+message ListContentRequest {
+  // Required. ID of the group to list content of.
+  string id = 1;
+
+  // Optional. Version of the group to list content of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of content to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing content from.
+  string page_token = 4;
+}
+
+// Response to list content.
+message ListContentResponse {
+  // List of content of the group.
+  // Key is the path and value is a Hyperdocs URL to the content.
+  map<string, string> content = 1;
+
+  // Token to continue listing content from.
+  string next_page_token = 2;
+}
+
+// Request to list groups.
+message ListGroupsRequest {
+  // Optional. Maximum number of groups to return.
+  int32 page_size = 1;
+
+  // Optional. Page token to continue listing groups from.
+  string page_token = 2;
+}
+
+// Response to list groups.
+message ListGroupsResponse {
+  // List of groups.
+  repeated Group groups = 1;
+
+  // Token to continue listing groups from.
+  string next_page_token = 2;
+}
+
+// Group represents the basic information about a group entity.
+message Group {
+  // ID of the group.
+  string id = 1;
+
+  // Title of the group.
+  string title = 3;
+
+  // Description of the group.
+  string description = 4;
+
+  // Account ID of the group owner.
+  string owner_account_id = 5;
+
+  // Timestamp when the group was first created.
+  google.protobuf.Timestamp create_time = 6;
+
+  // Latest version of the group according to the information from its owner and members
+  // that we happen to have locally. Most of the time this is the version that should be used
+  // to get the most recent group information.
+  string latest_version = 7;
+
+  // Latest version of the group according to our trusted contacts.
+  // It's possible that none of the group members is our trusted contact,
+  // but still some of our trusted contacts might mutate the group entity for themselves,
+  // in which case care must be takes while displaying the group entity to the user.
+  // If none of the group members is our trusted contact, this version will be empty.
+  // If only a subset of group members are our trusted contacts, this will be the head changes produced by them.
+  string latest_local_trusted_version = 8;
```

Comment:
They don't (using our software). We don't need to use it of course. It's here just because we can basically. No need to use it. I figured that it wouldn't cost me anything to add it here, and maybe eventually we might want to do something about it. Shouldn't hurt to have it.

Reply:
Again, the hurt is in the confusion of extra fields. Can we please remove fields that we don't intend on using?

Reply:
What should happen if we get independent changes from owner and editor, where editor published content, and owner removes the editor?

What if we get this retroactively even (we first get the change from editor, and only later the one from the owner which removes the editor making our previously applied change invalid)?

Reply:
I asked about this on one of our calls and you said that the owner will specify the last allowed group change from an editor when they revoke their permission. So it should be straightforward to determine which changes from the editor to ignore, no matter what order we get the changes.

But, whatever the solution is, the solution is used when computing the `latest_version`. This "trusted version" field still seems irrelevant and I think we should remove it.

Reply:
Yes, this approach of removing and editor can express which changes to trust. But it's weird when it happens retroactively, and I haven't realized it before. If we already applied the editor's change, now we have to "forget" about it, and there will be no way to expose that to the user, if we don't include these other version fields in the API.

But now I realize that we could implement a generic Entity API that would return all the change information, dependencies, and different versions for any entity, and then if we need we could use that one. So I'll remove all these other version fields.

Reply:
Great. I agree there are some tricky edge cases that will come up, but there are probably a lot more urgent things we haven't uncovered yet.